tests/unit/test_client.py

```diff
@@ -993,34 +993,103 @@ def test_create_bucket_w_object_success(self):
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(json_expected, json.loads(json_sent))
 
-    def test_download_blob_to_file_with_blob(self):
-        project = "PROJECT"
+    def test_download_blob_to_file_with_failure(self):
+        import requests
+        from google.resumable_media import InvalidResponse
+        from google.cloud import exceptions
+        from google.cloud.storage.blob import Blob
+
+        raw_response = requests.Response()
+        raw_response.status_code = http_client.NOT_FOUND
+        raw_request = requests.Request("GET", "http://example.com")
+        raw_response.request = raw_request.prepare()
+        grmp_response = InvalidResponse(raw_response)
+
+        blob_name = "blob-name"
+        media_link = "http://test.invalid"
         credentials = _make_credentials()
-        client = self._make_one(project=project, credentials=credentials)
-        blob = mock.Mock()
+        client = self._make_one(credentials=credentials)
+        bucket = _Bucket(client)
+        blob = Blob(blob_name, bucket=bucket)
+        blob._properties["mediaLink"] = media_link
+        blob._do_download = mock.Mock()
+        blob._do_download.side_effect = grmp_response
+
         file_obj = io.BytesIO()
+        with self.assertRaises(exceptions.NotFound):
+            client.download_blob_to_file(blob, file_obj)
 
-        client.download_blob_to_file(blob, file_obj)
-        blob.download_to_file.assert_called_once_with(
-            file_obj, client=client, start=None, end=None
+        self.assertEqual(file_obj.tell(), 0)
+
+        headers = {"accept-encoding": "gzip"}
+        blob._do_download.assert_called_once_with(
+            client._http, file_obj, media_link, headers, None, None, False
         )
 
-    def test_download_blob_to_file_with_uri(self):
-        project = "PROJECT"
+    def test_download_blob_to_file_wo_media_link(self):
+        from google.cloud.storage.blob import Blob
+
+        blob_name = "blob-name"
         credentials = _make_credentials()
-        client = self._make_one(project=project, credentials=credentials)
-        blob = mock.Mock()
+        client = self._make_one(credentials=credentials)
+        bucket = _Bucket(client)
+        blob = Blob(blob_name, bucket=bucket)
```

Comment:
We really don't need to care about whether the blob returns a media link or another path when testing the `Client` method. It would thus be better to use an autospec'ed mock here, e.g.:

```suggestion
        blob = mock.create_autospec(Blob)
```

and then mock out its `_get_download_url` method (as with the `_do_download` method).